### PR TITLE
Takes logic from core get_home_path

### PR DESCRIPTION
**Pull request by pdewouters**
I extracted the code from the core function `get_home_path` to use in our `get_home_path()` function

Should fix https://github.com/humanmade/backupwordpress/issues/976

### CODE REVIEW

**Comment by pdewouters** [comment]
looks like we need to change the unit tests

**Comment by willmot** [comment]
We'll need to mock siteurl and homeurl instead

**Comment by pdewouters** [comment]
For multisite, there seems to be some extra checks: https://github.com/WordPress/WordPress/blob/4a6d77c8e5fa846165cae3bb266298f284b43e15/wp-admin/includes/network.php#L334

**Comment by willmot** [comment]
Nice find, seems stupid that isn't rolled into `get_home_path`, but we might as well roll it into ours.

**Comment by pdewouters** [comment]
Here are the core unit tests for that function with some background https://core.trac.wordpress.org/ticket/20449

```
	function test_get_home_path() {
		$home = get_option( 'home' );
		$siteurl = get_option( 'siteurl' );
		$sfn = $_SERVER['SCRIPT_FILENAME'];
		$this->assertEquals( str_replace( '\\', '/', ABSPATH ), get_home_path() );

		update_option( 'home', 'http://localhost' );
		update_option( 'siteurl', 'http://localhost/wp' );

		$_SERVER['SCRIPT_FILENAME'] = 'D:\root\vhosts\site\httpdocs\wp\wp-admin\options-permalink.php';
		$this->assertEquals( 'D:/root/vhosts/site/httpdocs/', get_home_path() );

		$_SERVER['SCRIPT_FILENAME'] = '/Users/foo/public_html/trunk/wp/wp-admin/options-permalink.php';
		$this->assertEquals( '/Users/foo/public_html/trunk/', get_home_path() );

		$_SERVER['SCRIPT_FILENAME'] = 'S:/home/wordpress/trunk/wp/wp-admin/options-permalink.php';
		$this->assertEquals( 'S:/home/wordpress/trunk/', get_home_path() );

		update_option( 'home', $home );
		update_option( 'siteurl', $siteurl );
		$_SERVER['SCRIPT_FILENAME'] = $sfn;
	}
```

**Comment by pdewouters** [comment]
Is it bad practice to require a core file in this way?

**Comment by pdewouters** [comment]
why do we return the home path without the trailing slash?

**Comment by willmot** [comment]
> why do we return the home path without the trailing slash?

There's no good reason, nor is there a general convention either way that I'm aware of, although in WordPress I think only `ABSPATH` includes a trailing slash. I'm actually a fan of always including a trailing slash for reasons details in first answer here: http://stackoverflow.com/questions/17419460/should-i-save-a-path-with-or-without-a-trailing-slash-at-the-end-whats-the-con

**Comment by pdewouters** [comment]
wherever core did `str_replace( '\\', '/', ABSPATH )` , I replaced with `wp_normalize_path`, it seems to do the same thing but with extra checks

**Comment by pdewouters** [comment]
@willmot anything against merging this and releasing a beta?

File: classes/class-path.php [hunk 1]
```diff
@@ -105,18 +105,26 @@ public static function get_home_path( $site_path = ABSPATH ) {
...
-			if ( ( file_exists( dirname( $site_path ) . '/wp-config.php' ) || file_exists( $site_path . '/wp-config.php' ) )  && file_exists( dirname( $site_path ) . '/index.php' ) ) {
-				$home_path = dirname( $site_path );
+			$home    = set_url_scheme( get_option( 'home' ), 'http' );
+			$siteurl = set_url_scheme( get_option( 'siteurl' ), 'http' );
+			if ( ! empty( $home ) && 0 !== strcasecmp( $home, $siteurl ) ) {
+				$wp_path_rel_to_home = str_ireplace( $home, '', $siteurl ); /* $siteurl - $home */
+				$pos = strripos( wp_normalize_path( $_SERVER['SCRIPT_FILENAME'] ), trailingslashit( $wp_path_rel_to_home ) );
+				$home_path = substr( $_SERVER['SCRIPT_FILENAME'], 0, $pos );
+				$home_path = trailingslashit( $home_path );
+			} else {
```
Comment (willmot) [inline classes/class-path.php]: We don't need this else as `$home_path` is already set to `$site_path` above.

**Comment by willmot** [comment]
@pdewouters looks good, minor note about extraneous `else` statement but other than that happy for you to merge and release a beta 👍

**Pull request merged**